applications/crossbar/src/modules_v2/cb_phone_numbers_v2.erl

```diff
@@ -407,14 +408,68 @@ delete(Context, Number) ->
 %%--------------------------------------------------------------------
 -spec summary(cb_context:context(), ne_binary()) -> cb_context:context().
 summary(Context, Number) ->
+    IncludePorts = kz_util:is_true(
+                     cb_context:req_value(Context, <<"include_ports">>, 'false')
+                    ),
     Options = [{'auth_by', cb_context:auth_account_id(Context)}
               ],
     case knm_number:get(Number, Options) of
         {'ok', KNMNumber} ->
             crossbar_util:response(knm_number:to_public_json(KNMNumber), Context);
-        {'error', _JObj} -> reply_number_not_found(Context)
+        {'error', _JObj} -> maybe_find_port_number(Context, Number, IncludePorts)
     end.
 
+
+-spec maybe_find_port_number(cb_context:context(), ne_binary(), boolean()) ->
+                                    cb_context:context().
+maybe_find_port_number(Context, _Number, 'false') ->
+    reply_number_not_found(Context);
+maybe_find_port_number(Context, Number, 'true') ->
+    NormalizeNum = knm_converters:normalize(Number),
+    ViewOption = [{'key', [cb_context:account_id(Context), NormalizeNum]}],
+
+    case kz_datamgr:get_single_result(?KZ_PORT_REQUESTS_DB, ?PORT_NUM_LISTING, ViewOption) of
+        {'error', _} -> reply_number_not_found(Context);
+        {'ok', Result} ->
+            Port = kz_json:get_value(<<"value">>, Result),
+            {'ok', PhoneNumber} = normalize_port_number(Port, NormalizeNum, Context),
+            JObj = kz_json:set_values(
+                     [{[<<"_read_only">>, <<"port_id">>]
+                      ,kz_json:get_value(<<"port_id">>, Port)
+                      }
+                     ,{[<<"_read_only">>, <<"port_state">>]
+                      ,kz_json:get_value(<<"port_state">>, Port)
+                      }
+                     ], knm_phone_number:to_public_json(PhoneNumber)),
+            port_number_summary(JObj, Context, knm_phone_number:is_authorized(PhoneNumber))
+    end.
+
+%% @private
+-spec port_number_summary(kz_json:object(), cb_context:context(), boolean()) -> boolean().
+port_number_summary(PhoneNumber, Context, 'true') ->
+    crossbar_util:response(PhoneNumber, Context);
+port_number_summary(_PhoneNumber, Context, 'false') ->
+    reply_number_not_found(Context).
+
+-spec normalize_port_number(kz_json:object(), ne_binary(), cb_context:context()) ->
+                                   knm_phone_number:knm_phone_number().
+normalize_port_number(JObj, Number, Context) ->
+    Features = lists:foldl(fun (FeatureKey, Acc) ->
```

Comment:
There is knm_phone_number:set_feature/3 (which may need setters/2 to be updated to support 3-tuples).
I would prefer if that translation between list to jobj only happened in KNM